demos/TicTacToe/Classes/TicTacToeScene.cpp

```diff
@@ -0,0 +1,81 @@
+#include "TicTacToeScene.h"
+
+#include <array>
+#include <cstdlib>
+#include <iostream>
+
+#include "cocos2d.h"
+
+USING_NS_CC;
+
+const int kNumberOfTiles = 9;
+const int kMaxMovesPerPlayer = 1 + kNumberOfTiles / 2;
+const int kTilesX = std::sqrt(kNumberOfTiles);
```

Comment:
I would do this the other way:

kTilesX = 3
kTilesY = 3
kNumberOfTiles = kTilesX * kTilesY;